src/gui/sdlmain.cpp

```diff
@@ -3936,6 +3937,45 @@ int sdl_main(int argc, char *argv[])
 	const auto config_path = CROSS_GetPlatformConfigDir();
 	SETUP_ParseConfigFiles(config_path);
 
+	MSG_Add("PROGRAM_CONFIG_PROPERTY_ERROR", "No such section or property.\n");
+	MSG_Add("PROGRAM_CONFIG_NO_PROPERTY",
+		"There is no property \"%s\" in section \"%s\".\n");
+	MSG_Add("PROGRAM_CONFIG_SET_SYNTAX",
+		"Correct syntax: config -set \"section property\".\n");
+	std::string line;
+	while (control->cmdline->FindString("-set", line, true)) {
+		// add rest of command
+		std::string rest;
+		std::vector<std::string> pvars;
+		trim(line);
+		pvars.clear();
+		pvars.push_back(line.c_str());
```

Comment:
I can't see where `rest` is used - suggest dropping it.
`pvars.clear()` isn't needed because `std::vector<std::string> pvars` is a new object.
`pvars.push_back(line.c_str());`: pvars already wants to hold actual `std::string` object.
So by converting line back to a C char* array, we then need to contstruct a new string and re-copy those elements (instead of just passing the string-object itself).  On top of that, pushing back then constructs a new object inside the vector and uses the assignment operator - copying the string again.

Reply:
vector's can be constructed straight away with content, and if the content is a complex object (like std::string), `std::move` can be used to "transfer the content" into the vector, and hollow-out the object, all without any copying being done. (example below)

Reply:
rest was from program.cpp, which is indeed unnecessary here. Also, there is indeed no need for c_str() for putting into the pvars vector.